test/unit/write_concern.test.ts

```diff
@@ -0,0 +1,124 @@
+import { expect } from 'chai';
+
+import { WriteConcern } from '../mongodb';
+
+describe('WriteConcern', function () {
+  describe('#constructor', function () {
+    context('when w is provided', function () {
+      context('when w is a number', function () {
+        const writeConcern = new WriteConcern(1);
+
+        it('sets the w property', function () {
+          expect(writeConcern.w).to.equal(1);
+        });
+      });
+
+      context('when w is a string number', function () {
+        const writeConcern = new WriteConcern('10');
+
+        it('sets the w property to a number', function () {
+          expect(writeConcern.w).to.equal(10);
+        });
+      });
+
+      context('when w is a string', function () {
```

Comment:
Do we have any behaviour defined for when `w` is an arbitrary string? I can see it being a possibility that people consuming the library from regular JavaScript that we silently do something different from what they expected if they were to make a typo like `'majorit'` instead of `'majority'`.

Reply:
The server will return a write concern error if it is a string and not 'majority'.